server/src/main/java/io/mantisrx/master/jobcluster/job/worker/MantisWorkerMetadataImpl.java

```diff
@@ -186,6 +185,41 @@ public int getNumberOfPorts() {
         return Optional.ofNullable(workerPorts);
     }
 
+    /**
+     * Attempts to build a list of port numbers.
+     *
+     * The difference between this method and simply returning an empty list is that this will at least pass in *some*
+     * value for a port which can be caught by the caller to handle if the caller determines it's invalid.
+     *
+     * Previously, we had code like this:
+     *
+     * workerHosts.put(wm.getWorkerNumber(),
+     *     new WorkerHost(wm.getSlave(), wm.getWorkerIndex(),
+     *             wm.getPorts().isPresent() ? wm.getPorts().get().getPorts()
+     *                     : Lists.newArrayList(),
+     *             DataFormatAdapter.convertWorkerStateToMantisJobState(wm.getState()),
+     *             wm.getWorkerNumber(),
+     *             wm.getMetricsPort(),
+     *             wm.getCustomPort()));
+     *
+     * which would result in issues like an array index out of bounds.
+     *
+     * @return a list of port numbers from the {@link WorkerPorts} if possible, otherwise manually builds the list.
+     */
+    @JsonIgnore
+    public List<Integer> getPortList() {
+        if (getPorts().isPresent()) {
+            return getPorts().get().getPorts();
+        }
+
+        return Lists.newArrayList(
```

Comment:
Building this list exactly as it previously was, just moved this method to this spot.

Reply:
since WorkerMetadata already has a getWorkerPorts(), this method feels redundant. The calling code can derive this info from getWorkerPorts()?

Reply:
I think I can move this out of this class. The important thing is that we don't return an empty list as we did before https://github.com/Netflix/mantis-control-plane/pull/26/files#diff-81433f6ab0cca184eca4716606d9b3eeL1448-L1455. From my understanding, we don't have ports in the data structure initially. Port info is filled in afterwards, so methods like `wm.getMetricsPort()` return a `-1` if worker ports don't exist. I can move this out and build a list of `-1`s in `JobActor`. thoughts?